### PR TITLE
HDDS-10701. Create config option for keystores

### PR DESCRIPTION
**Pull request by Galsza**
## What changes were proposed in this pull request?
Keystores by default use the "jks" type and it's not configurable currently. After investigation keyStore.getDefaultType() can be used here, as it is possible to change that via Security.setProperty() or through config files.

This effort is part of a larger effort for crypto compliance where each cryptographic algorithm should be marked with the CRYPTO_COMPLIANCE tag so that later a whitelisting mechanism can be implemented for them based on this tag.

https://issues.apache.org/jira/browse/HDDS-10701

## How was this patch tested?
No functional code changes
Green CI (before the whitelist change in ozone-default.xml): 
https://github.com/Galsza/ozone/actions/runs/8718309829

### CODE REVIEW

**Review by fapifta — COMMENTED** [review]
I have a general doubt and I am not sure if we need to deal with this much around configuring keystore type for an in-memory KeyStore.
That keystore type in this case is not really utilized much in any context as I understand, and it would have only any meaning if we load the keystore from some stream, and then we store it to some stream, but in this case we load the certificates separately from a PEM file, and we put the certificate object to these keystores.

I'd argue to create an in-memory keystore instead, where we disable the load from anything but (null, null) as we do now, and we also disable store, and we somehow hook that modified keystore here, with type in-mem and provider maybe set to the default provider in the system. (This latter might be needed for JSSE when fips mode is enabled, as in this case it validates the provider in the keystore against the crypto provider set in JSSE as I understand that code.

What do you think @Galsza?

**Comment by Galsza** [comment]
@fapifta thank you for your comment and yes, I agree with changing the functionality here. Not sure just yet how I'll implement it, but I'll reach out to you if there are questions

**Review by fapifta — COMMENTED** [review]
Based on our discussion with @Galsza the best choice as it seems is to use the Keystore.getDefaultType() method to get the type for this in-memory Keystore, and then forget about this, as we do not directly use the created Keystore instance later on, and also because the default type is configurable, and is expected to be configured by the compliant security provider to a compliant default. This is to be tested on a per provider basis though, and in case it does not set the property, still this can be set via the java.security file.

**Comment by Galsza** [comment]
@fapifta Thanks for the suggestion, I've updated the code and the pr description.

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/security/ssl/PemFileBasedKeyStoresFactory.java [hunk 1]
```diff
@@ -51,7 +51,7 @@ public class PemFileBasedKeyStoresFactory implements KeyStoresFactory,
   /**
    * Default format of the keystore files.
    */
-  public static final String DEFAULT_KEYSTORE_TYPE = "jks";
+  public static final String DEFAULT_KEYSTORE_TYPE = KeyStore.getDefaultType();
```
Comment (fapifta) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/security/ssl/PemFileBasedKeyStoresFactory.java:54]: As I see we use the constant only internally in this class, can we either inline this, or make it private? I prefer inlining the call, even though it means calculating the value every time, but the trust/keystores are cached so it does not happen often, but help us to see all the usages right away when we check for this method within the code

**Review by fapifta — COMMENTED** [review]
Thank you for the changes @Galsza please take a look to potentially one last comment inline.

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/security/ssl/PemFileBasedKeyStoresFactory.java (hunk 1 above)
Comment (Galsza) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/security/ssl/PemFileBasedKeyStoresFactory.java]: I've inlined the usages. I agree with it being easier to search that way, which might or might not come in handy later.

**Review by fapifta — APPROVED** [review]
Thank you for the changes @Galsza! LGTM.

**Pull request merged**